### ISSUE TITLE
can't overwrite 'jwt.secret' using profile 'native' or command line arg

### ISSUE THREAD

**Issue by tkirsch** [jhipster/jhipster-registry#281]
##### **Overview of the issue**

while using 'native' in 'spring.profiles.active' we can't overwrite the default jwt.secret provided by the https://github.com/jhipster/jhipster-registry-sample-config
using the command line arg doesn't work either.
it does work using our own git repo.

##### **Motivation for or Use Case**

not working as expected and described in the documentation 'Using the pre-packaged WAR file'.

##### **Reproduce the error**

create ./central-config with application.yml containing the jwt.secret.
start the jhipster-registry.war and check Configuration > Cloud Config. 

##### **JHipster Registry Version(s)**

3.3.0

##### **Browsers and Operating System**

Win 10
java 1.8.0_66
node v8.11.3
npm 6.1.0

- [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by ruddell** [comment]
Can confirm, not sure of the solution.  Maybe related to the [bootstrap](https://github.com/jhipster/jhipster-registry/blob/master/src/main/resources/config/bootstrap.yml#L14-L17) config having both git/native?

**Comment by jdubois** [comment]
I'm sorry I can't test right now -> do you mean that even with the native profile it still fetches the default Git config? That would be pretty bad.
As this is configured in the bootstrap.yml, yes that's probably the culprit.

**Comment by PierreBesson** [comment]
No the native profile is correctly configured but the registry doesn't setup the jwt secret in its own properties from central config/application.yml or perhaps this is caused by the code you added at boot time.

No the problem is not due to the git config in bootstrap.yml this is safely ignored due to the native profile.

**Comment by pascalgrimaud** [comment]
Sorry, I should post here instead of our ML. So copy/paste:

To reproduce:
- download the v3.3.0 at https://github.com/jhipster/jhipster-registry/releases into an empty folder
- go into this folder, then launch: `java -jar jhipster-registry-3.3.0.war --spring.profiles.active=dev,native,swagger --jhipster.security.authentication.jwt.secret=toto`
- go into config screen and you will see:

```yaml
configserver:
  name: JHipster Registry config server
  status: Connected to the JHipster Registry config server, using https://github.com/jhipster/jhipster-registry-sample-config
    !
jhipster:
  security:
    authentication:
      jwt:
        secret: my-secret-token-to-change-in-production
```

It shouldn't fetch the config from https://github.com/jhipster/jhipster-registry-sample-config as I use native profile.
You can look the response times too.

I did the same with v3.2.4 and it works well, without any issue. So something has changed... Maybe the upgrade of Spring Cloud ?

**Comment by PierreBesson** [comment]
I really don't understand what is going on here and why it doesn't work. My best guess is an issue related to the native profile: https://github.com/spring-cloud/spring-cloud-config/issues/897.
The weird issue, is that when I remove the `git.uri` properties, it complains that it is missing even though I'm using the `native` profile. On my local testing, the native profile is completely broken.
An alternative would be to set the JWT key with :
```
spring:
    cloud:
        config:
            server:
                overrides:
                    jhipster:
                        security:
                            authentication:
                                jwt:
                                    secret: set-this-secret-on-the-registry-with-env-vars
```

**Comment by pascalgrimaud** [comment]
@PierreBesson : can you try this change in `bootstrap.yml` file ?

```diff
diff --git a/src/main/resources/config/bootstrap.yml b/src/main/resources/config/bootstrap.yml
index 11f1009..e751457 100644
--- a/src/main/resources/config/bootstrap.yml
+++ b/src/main/resources/config/bootstrap.yml
@@ -11,12 +11,9 @@ spring:
     cloud:
         config:
             server:
-                git:
-                    uri: https://github.com/jhipster/jhipster-registry-sample-config
                 native:
                     search-locations: file:./central-config
                 prefix: /config
-                bootstrap: true
             fail-fast: true
             # name of the config server's property source (file.yml) that we want to use
             name: jhipster-registry
```

**Comment by PierreBesson** [comment]
@pascalgrimaud if you remove the bootstrap property, it won't be able to retrieve configuration from itself... But yes it does fix our issue with the native profile (which I suspect is a spring cloud config bug) but it's not totally satisfactory. Let's discuss this tomorrow with @jdubois.

**Comment by pascalgrimaud** [comment]
Yes, I think this ticket must be our priority tomorrow :)

**Comment by pascalgrimaud** [comment]
fixed by #282
